### ISSUE TITLE
Month not updated when DatePicker is in MaterialTable

### ISSUE THREAD

**Issue by mosofsky** [mui/mui-x#8281]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example: [Month not updated when DatePicker is in MaterialTable](https://codesandbox.io/s/react-typescript-playground-forked-tikojq?file=/src/index.tsx:2560-2573)

Video reproduction: [Month not updated when DatePicker is in MaterialTable](https://youtu.be/D7NakOSO8l8)

Steps:
1. Click the pencil icon on the row with the date 2023-03
2. The date field should become editable and the date should render in the format 03/2023
3. Click the calendar icon
4. Click 2017
5. Click July
6. The date field should have changed from 03/2023 to 07/2017 but instead it says 03/2017 (i.e. the month did not update)

### Current behavior 😯

The date changes from 03/2023 to 03/2017 (i.e. the month did not update)

### Expected behavior 🤔

The date should change from 03/2023 to 07/2017 (i.e. the month should update)

### Context 🔦

Trying to let users enter a month and year representing when something needs to be manufactured.

### Your environment 🌎

See [package.json])https://codesandbox.io/s/react-typescript-playground-forked-tikojq?file=/package.json)

**Comment by LukasTy** [comment]
Hello @mosofsky 
You have an issue in your `onAccept` prop. The function uses `props.value` and hence resets the date value when executed.
From what I've tested—the prop did not seem as if it was needed at all, but I might be mistaken.
Removing the `onAccept` prop or replacing it with the following change fixes your issue.
```diff
- onAccept={() => handleManufactureDateBlur(props)}
+ onAccept={(value) => handleManufactureDateBlur({ ...props, value })}
```

**Comment by mosofsky** [comment]
I can still reproduce the issue after eliminating onAccept.  I'm just using a straightforward implementation now:

```
                <DatePicker
                    slotProps={{ textField: { id: "idEndDate" } }}
                    value={selectedDate}
                    onChange={(date) => handleEndDateOnChange(date, props)}
                    views={[DATE_PART_YEAR, DATE_PART_MONTH]}
                    format={MM_YYYY}
                />
```

It seems that the DatePicker doesn't fire onChange when placed too close to the bottom of the screen.  This video demonstrates that as the DatePicker gets closer to the bottom of the screen, it's not able to fire onChange:

1. Video: https://youtu.be/_ugUfu0DSJw
2. Playground with the code: https://codesandbox.io/s/react-typescript-playground-forked-tikojq?file=/src/index.tsx

Here is another repro but I can't share the code: https://youtu.be/nW83TZwjnxc

**Comment by mosofsky** [comment]
@LukasTy can you reopen this ticket?  Two months ago I commented that the problem is reproducible even after I corrected the syntax of my `onAccept`.  I provided a Playground with the code to reproduce.

**Comment by LukasTy** [comment]
@mosofsky Thank you for reminding me about this. 🙏 
The underlying problem seems to be the same as in #5490.
There is a possible workaround suggested in the mentioned issue, that you could try using before we decide on the most correct global solution. 😉 

P.S. I believe that you should also be able to reopen this issue. 🤔

**Comment by mosofsky** [comment]
@LukasTy actually I am unable to change the status from Closed back to Open.  Maybe I don't have enough permission.  Would still love to see a fix for this because I can't upgrade and I'm stuck on "@mui/x-date-pickers": "^5.0.20".

**Comment by LukasTy** [comment]
Hello @mosofsky 
Based on the proposed solution on https://github.com/mui/mui-x/pull/10541 you can easily apply the "same" remedy on your end using styling approach that best suits you.
For example, [here is a solution](https://codesandbox.io/s/mui-picker-popover-forked-s6r8f3?file=/src/App.tsx) with an applied fix, based on the issue that PR targets to solve.
Let me know if that helps you solving your issue on v5. 🙏 

We no longer actively support v5 and this does not seem like a critical issue, especially when it can be remedied quite easily with externally applying same/similar styling fix.

**Comment by mosofsky** [comment]
As of "@mui/x-date-pickers": "^6.18.6" I am no longer experiencing this problem so I don't need the workaround.  Thanks developers!

**Comment by LukasTy** [comment]
@mosofsky To be precise, the fix has been released in `6.16.1`. 😉
